### PR TITLE
Throttle or block mishbehaving peers.

### PR DESCRIPTION
**Pull request by noffle**

### CODE REVIEW

File: swarm.js [hunk 1]
```diff
@@ -3,22 +3,47 @@ var discovery = require('discovery-swarm')
...
-      conn.once('end',   function () { if (remoteKey) cabal._removeConnection(remoteKey) })
+      if (blocked[remoteKey]) return
+      blocked[remoteKey] = true
+      connected[remoteKey] = connected[remoteKey] ? connected[remoteKey]+1 : 1
 
       var r = cabal.replicate()
       pump(conn, r, conn, function (err) {
         if (err) debug('ERROR', err)
+
+        cabal._removeConnection(remoteKey)
```
Comment (cblgh) [inline swarm.js]: why do we do this part?

**Comment by noffle** [comment]
This tells cabal-core to emit a `peer-removed` event, which the client listens to in order to remove peers from the right sidebar as active.

**Comment by todrobbins** [comment]
Have we considered muting as well?

**Comment by todrobbins** [comment]
PS: I like how you've approached block times. ✌🏻

**Comment by noffle** [comment]
@todrobbins: what do you mean by muting?

**Comment by todrobbins** [comment]
It's analogous, but maybe something similar to Twitter's approach: https://help.twitter.com/en/using-twitter/twitter-mute

So I might not want to have a certain user's messages/files ending up in my feed/channel view, but may want to mute their posting. This might be useful for blocking content I find distasteful, but may not violate the community standards of any particular cabal.

Does that make sense?

**Comment by todrobbins** [comment]
See also: https://www.guidingtech.com/twitter-block-vs-mute-difference/

Muting has been interpreted as a less severe version of blocking. 🤷🏻‍♂️

**Comment by noffle** [comment]
In this PR "block" just means "don't allow connections from this peer for the rest of the time this cabal instance is running". It's not about their content, so I'm not sure what "muting" would mean in this context.

**Comment by noffle** [comment]
Merged. Github didn't pick it up?

**Pull request closed without merging**